src/repository.py

```diff
@@ -280,6 +280,7 @@ def is_commit_in_branch(self, commit_sha: str, branch: str | None = None) -> boo
         """
         star_pattern = re.compile(r"^\* ")
         try:
+            self._git_repo.git.fetch("--depth=2147483647")
```

Comment:
> The special depth 2147483647 (or 0x7fffffff, the largest positive number a signed 32-bit integer can contain) means infinite depth.

Reference [here](https://git-scm.com/docs/shallow)

Reply:
Perhaps make this a comment on the code?